plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend

```diff
@@ -157,29 +157,35 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
 		checkDuplicated(message, model as Model, newHashSet)	
 	}
 	
-	def boolean checkDuplicated(Message message, EObject model, HashSet<String> names) {
+	def boolean checkDuplicated(EObject elementToCheck, EObject model, HashSet<String> names) {
 		var elements = TextualScenarioHelper.getElements(model)
 		for (element : elements) {
 
-			if (element instanceof SequenceMessageType || element instanceof ArmTimerMessage) {
-				if (!names.add(getMessagesMapKey(element)) && element.equals(message)) {
+			if (element instanceof SequenceMessageType || element instanceof ArmTimerMessage ||
+				element instanceof CombinedFragment) {
+				if (!names.add(getElementMapKey(element)) && element.equals(elementToCheck)) {
 					if (element instanceof SequenceMessageType) {
 						var source = (element as SequenceMessageType).source
 						var target = (element as SequenceMessageType).target
-						error('The same exchange is already used in text editor between \"' +
-							source + "\" and \"" + target + "\"!",
-							TextualScenarioPackage.Literals.MESSAGE__NAME)
-					} else if(element instanceof ArmTimerMessage) {
+						error(
+							'The same exchange is already used in text editor between \"' + source + "\" and \"" +
+								target + "\"!", TextualScenarioPackage.Literals.MESSAGE__NAME)
+					} else if (element instanceof ArmTimerMessage) {
 						error('The same exchange is already used in text editor on timeline \"' +
-							(element as ArmTimerMessage).participant +"\"!",
+							(element as ArmTimerMessage).participant + "\"!",
 							TextualScenarioPackage.Literals.MESSAGE__NAME)
+					} else if (element instanceof CombinedFragment) {
+						error(String.format(
+							"The same " + element.keyword + " with expression \" " + element.expression +
+								"\" and timelines " + element.timelines.toSet.sort + " is already used in text editor!"
+						), TextualScenarioPackage.Literals.COMBINED_FRAGMENT__EXPRESSION)

```

Comment:
No need to sort, just element.timelines.

Reply:
done